.github/workflows/cd-publish-docs-branch.yml

```diff
@@ -0,0 +1,82 @@
+---
+# yamllint disable line-length
+name: "Publish docs to GitHub Pages"
+
+on: # yamllint disable-line rule:truthy
+  workflow_dispatch:
+  repository_dispatch:
+    types: [ci-publish-docs-branch]
```

Comment:
I don't think this will work the way it should. This (along with the change to ci-publish-docs-branch.yml) will cause the CD to occur as soon as the CI is done. This is not what is desired. We want to trigger the CD when the docs branch is merged into main. That won't happen until after the PR that is automatically generated by ci-publish-docs-branch is approved and merged.

Take a look at `cd-create-version-tag.yml` specifically:

```
on: #  yamllint disable-line rule:truthy
  push:
    branches: [main]
  pull_request:
    branches: [main]

jobs:
  create-version-tag:
    if: ${{ github.event_name == 'push' && github.event.pull_request.head.ref == 'refs/heads/version' }}
```

This will trigger whenever a PR action is taken on main, or a push to main happens. The `if` block limits it to only when it is a PR that caused the push to main from the `version` branch.

Reply:
Definitely agree with this @marclerwick  - However we wouldn't merge into main. docs would be a separate branch as it hosts the GitHub pages site. All this still applies, but it would be merging a feature branch (with docs changes) to docs.

Note this is for the autogenerated reference docs, not that markdown written in the docs folder. In the future we will have to consolidate both.

Reply:
@hattan That's not what `ci-publish-docs-branch.yml` does though. `ci-publish-docs-branch.yml` cuts the `docs` branch from the `main` branch, then generates the autogenerated documents. Then, if there is a diff on those docs that are generated from the existing it will commit and push those diffs on the `docs` branch. And lastly, a PR is automatically created to from `docs` to `main`.